### PR TITLE
shader_jit: Load LOOPCOUNT_REG and LOOPINC 4 bit left-shifted

### PR DESCRIPTION
**Pull request by JayFoxRox**
As noticed by @Subv and pointed out by @yuriks in https://github.com/citra-emu/citra/issues/1629#issuecomment-263884563 , our shader-JIT had some serious issue where the LOOP address register was used directly in offset calculations.
As each addressed vector is 16-bytes in size, this PR changes the JIT registers by keeping their values shifted 4 bits to the left (effectively keeping these regs multiplied by 16 at all times).

The offset used in address calculations is stored in LOOPCOUNT_REG. It's incremented by LOOPINC.
Hence both registers were shifted.

---

This fixes #1629 and #2226.

Additionally the indexed lookup is performed using MOVAPS which requires the memory address to be 16 byte aligned. As we didn't satisfy that condition with the old code, we probably read garbage. This might have been responsible for #1743 and #2209 (Issues on certain AMD Phenom CPUs).

---

TODO:

- [x] Cleanup
- [ ] PR Description
- [ ] Testing

### CODE REVIEW

**Review by Subv — APPROVED** [review]
LGTM

File: src/video_core/shader/shader_jit_x64.cpp [hunk 1]
```diff
@@ -718,15 +718,19 @@ void JitShader::Compile_LOOP(Instruction instr) {
 
     looping = true;
 
+    // This block of code decodes the fields from the integer uniform at
+    // index instr.flow_control.int_uniform_id
+    // The Y (LOOPCOUNT_REG) and Z (LOOPINC) component are kept multiplied by 16 (Left shifted
+    // by 4 bit) to be used as an offset into the 16-byte vector registers later
```
Comment (yuriks) [inline src/video_core/shader/shader_jit_x64.cpp]: Pretty wonky and inconsistent word-wrapping in this comment.

**Review by yuriks — APPROVED** [review]

File: src/video_core/shader/shader_jit_x64.cpp (hunk 1 above)
Comment (JayFoxRox) [reply src/video_core/shader/shader_jit_x64.cpp]: Yes, I've changed it a couple of times already but this is the best I could come up with :P

Comment (yuriks) [reply src/video_core/shader/shader_jit_x64.cpp]: At least put a full stop after "int_uniform_id" then. But I don't think these have to be 2 separate paragraphs.

Comment (JayFoxRox) [reply src/video_core/shader/shader_jit_x64.cpp]: It's supposed to be a full stop [in the form of a new line]. I've had `instr.flow_control.int_uniform_id.` before but it just looks unintended due to the many dots in the field name itself.

Comment (Dragios) [reply src/video_core/shader/shader_jit_x64.cpp]: Then just use full stop followed by double spacing or alternatively |

Comment (yuriks) [reply src/video_core/shader/shader_jit_x64.cpp]: +1 on just use a period and then double-spacing.

**Comment by JayFoxRox** [comment]
Moved to main repo & accepted

**Pull request closed without merging**